src/go/rpk/pkg/cli/cmd/topic/autorestore/start.go

```diff
@@ -0,0 +1,104 @@
+// by the Apache License, Version 2.0
+package autorestore
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newStartCommand(fs afero.Fs) *cobra.Command {
+	var topicNamePattern string
+	var wait bool
+	var pollingInterval uint
+
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Start the autorestore process",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			client, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			ctx := cmd.Context()
+
+			_, err = client.StartAutomatedRecovery(ctx, topicNamePattern)
+			var he *admin.HTTPResponseError
+			if errors.As(err, &he) {
+				if he.Response.StatusCode == 404 {
+					body, bodyErr := he.DecodeGenericErrorBody()
+					if bodyErr == nil {
+						out.Die("Not found: %s", body.Message)
+					}
+				} else if he.Response.StatusCode == 400 {
+					body, bodyErr := he.DecodeGenericErrorBody()
+					if bodyErr == nil {
+						out.Die("Cannot start auto-restore: %s", body.Message)
+					}
+				}
+			}
+
+			out.MaybeDie(err, "error starting auto-restore: %v", err)
+			fmt.Println("Successfully started auto-restore")
+
+			if !wait {
+				return
+			}
+			pollingIntervalSeconds := time.Duration(pollingInterval) * time.Second
+
+			fmt.Println("Waiting for auto-restore to complete...")
+
+			for {
+				status, err := client.PollAutomatedRecoveryStatus(ctx)
+				out.MaybeDie(err, "failed to poll automated recovery status: %v", err)
+
+				pending := false
+				for _, topicDownload := range status.TopicDownloads {
+					if topicDownload.PendingDownloads > 0 {
+						pending = true
+						break
+					}
+				}
+
+				if !pending {
+					failedPartitionReplicas := []string{}
+					for _, topicDownload := range status.TopicDownloads {
+						if topicDownload.FailedDownloads > 0 {
+							failedPartitionReplicas = append(failedPartitionReplicas, topicDownload.TopicNamespace)
+						}
+					}
+
+					if len(failedPartitionReplicas) > 0 {
+						out.Die("automated recovery failed to download partition replicas: %v", failedPartitionReplicas)
+					}
+
+					break
+				}
+				time.Sleep(pollingIntervalSeconds)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&topicNamePattern, "topic-name-pattern", ".*", "A regex pattern to match topic names against. Only topics whose names match this pattern will be restored. If not passed, all topics will be restored.")
+	cmd.Flags().BoolVarP(&wait, "wait", "w", false, "Wait until auto-restore is complete.")
+	cmd.Flags().UintVar(&pollingInterval, "polling-interval", 5, "The interval in seconds between each status check. Ignored if --wait is not passed.")
```

Comment:
_I think_ there is a duration flag: `cmd.Flags().DurationVar` so you don't have to parse and convert to duration in the Run section of the command.

Reply:
After looking into this I think `cmd.Flags().DurationVar` only accepts input as nanoseconds, which I think is impractical from a UX perspective as the user will have to add a lot of zeros every time they want to use a different interval

Reply:
I looked into this, and see how it's used by rpk, example: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/cmd/debug/bundle/bundle.go#L164

It accepts durations like '1m', '2h', etc ...  basically whatever it's accepted by go std time.ParseDuration

Reply:
Thanks!

That was not clear from the documentation that it would work like that :grimacing: